packages/components/src/components/domainproperties/dataset/DatasetPropertiesPanelFormElements.tsx

```diff
@@ -0,0 +1,280 @@
+                    name={radioName}
+                    value={0}
+                    checked={dataRowSetting == 0}
+                    onChange={onRadioChange}
+                >
+                    Participant ID only (demographic data)
+                </Radio>
+
+                <Radio
+                    name={radioName}
+                    value={1}
+                    checked={dataRowSetting == 1}
+                    onChange={onRadioChange}
+                >
+                    Participant ID and timepoint
+                </Radio>
+
+                <Radio
+                    name={radioName}
+                    value={2}
+                    checked={dataRowSetting == 2}
+                    onChange={onRadioChange}
+                >
+                    Participant ID, timepoint, and additional key field
+                </Radio>
+            </div>
+        );
+    }
+}
+
+interface DataRowUniquenessContainerProps {
+    model: DatasetModel;
+    onRadioChange: (e: any) => any;
+    onCheckBoxChange: (any) => void;
+    onSelectChange: (name, formValue, selected) => void;
+}
+
+export class DataRowUniquenessContainer extends React.PureComponent<DataRowUniquenessContainerProps> {
+
+    getDataRowSetting(model: DatasetModel) : number {
+        let dataRowSetting;
+
+        // participant id
+        if ((model.keyPropertyId === undefined || model.keyPropertyId === null) && model.isDemographicData) {
+            dataRowSetting = 0;
+        }
+        // participant id and timepoint
+        else if (model.keyPropertyId === undefined || model.keyPropertyId === null) {
+            dataRowSetting = 1;
+        }
+        // participant id, timepoint and additional key field
+        else {
+            dataRowSetting = 2;
+        }
+
+        return dataRowSetting;
+    }
+
+    getHelpTipElement(field: string) : JSX.Element {
+        return <> {getHelpTip(field)} </> as JSX.Element;
+    }
+
+    render() {
+        const { model, onRadioChange, onCheckBoxChange, onSelectChange } = this.props;
+        const domain = model.domain;
+
+        const dataRowSetting = this.getDataRowSetting(model);
+        const showAdditionalKeyField = dataRowSetting === 2;
+
+        const showAdditionalKeyFieldCls = showAdditionalKeyField ? "dataset_data_row_uniqueness_keyField_show margin-top" : "dataset_data_row_uniqueness_keyField_hide";
+
+        return (
+            <>
+                <SectionHeading
+                    title="Data Row Uniqueness"
+                    helpTipBody={() => this.getHelpTipElement("dataRowUniqueness")}
+                />
+
+                <DataRowUniquenessElements
+                    onRadioChange={onRadioChange}
+                    dataRowSetting={dataRowSetting}
+                />
+
+                <div className={showAdditionalKeyFieldCls}>
+                    <DatasetSettingsSelect
+                        name={"keyPropertyId"}
+                        label={"Additional Key Field"}
```

Comment:
for the help tip to use on this label, see the GWT version on "Additional Key Column". See also in that help text where it talks about the managed field checkbox only applying to numeric or string fields, have you added that check?

Reply:
nope, was planning to filter the fields in the next story. But can do now as well.

Reply:
Next story is fine for the field type check, maybe just add a TODO in the code for now.